demos/adjoint/derivatives.py

```diff
@@ -0,0 +1,158 @@
+from gadopt import *
```

Comment:
Yeah, that's a good idea. I'm actually not sure what this `derivatives.py` one does, might be one of @sghelichkhani's tests that doesn't need to go into the final repo? I think `inverse.py` and `inverse_checkpoint.py` should be consolidated, because they realistically only differ by one line!

Also, for structuring, maybe `helmholtz.py` should go in its own directory, since it's more a demonstration/test of the checkpointing..?